lib/logstash/inputs/elasticsearch.rb

```diff
@@ -196,8 +205,25 @@ def stop
   private
 
   def do_run(output_queue)
-    # get first wave of data
-    r = @client.search(@options)
+    # if configured to run a single slice, don't bother spinning up threads
+    return do_run_slice(output_queue) if @slices.nil? || @slices <= 1
+
+    logger.warn("managed slices for query is very large (#{@slices}); consider reducing") if @slices > 8
+
+    @slices.times.map do |slice_id|
+      Thread.new do
```

Comment:
let's name these threads so in the future we'll be able to identify them easily in a thread dump